src/components/scenes/FioRequestConfirmationScene.js

```diff
@@ -0,0 +1,204 @@
+          memo: this.props.fioModalData.memo,
+          maxFee: 0
+        })
+        this.setState({ loading: false })
+        showToast(s.strings.fio_request_ok_body)
+        Actions.popTo(Constants.TRANSACTION_LIST)
+      } catch (error) {
+        this.setState({ loading: false })
+        showError(`${s.strings.fio_request_error_header}. ${error.json ? JSON.stringify(error.json.fields[0].error) : ''}`)
+      }
+    }
+  }
+
+  fiatAmount = (amount: string): string => {
+    const fiatPerCrypto = this.props.exchangeSecondaryToPrimaryRatio
+    return intl.formatNumber(fiatPerCrypto * parseFloat(amount), { toFixed: 2 }) || '0'
+  }
+
+  handleFioWalletChange = (something: string, index: number): void => {
+    this.setState({ selectedFioAddress: this.state.walletAddresses[index].fioAddress })
+  }
+
+  labelFromItem = (item: { fioAddress: string, label: string }): string => {
+    return item.label
+  }
+
+  render () {
+    const { loading, selectedFioAddress, walletAddresses } = this.state
+    const { primaryCurrencyInfo, secondaryCurrencyInfo, exchangeSecondaryToPrimaryRatio } = this.props
+    if (!primaryCurrencyInfo || !secondaryCurrencyInfo) return null
+    let finalValue, val
+    try {
+      finalValue = bns.div(this.props.amounts.nativeAmount, primaryCurrencyInfo.displayDenomination.multiplier, 18)
+      val = bns.div(this.props.amounts.nativeAmount, primaryCurrencyInfo.exchangeDenomination.multiplier, 18)
+    } catch (e) {
+      return null
+    }
+    const fiat = this.fiatAmount(val)
+    const MaterialInputStyle = {
+      ...MaterialInput,
+      container: {
+        ...MaterialInput.container,
+        width: styles.input.width
+      }
+    }
+
+    return (
+      <SceneWrapper>
+        <View style={styles.exchangeRateContainer}>
+          <ExchangeRate primaryInfo={primaryCurrencyInfo} secondaryInfo={secondaryCurrencyInfo} secondaryDisplayAmount={exchangeSecondaryToPrimaryRatio} />
+        </View>
+        <View style={styles.textContainer}>
+          <T style={styles.text}>
+            {s.strings.fio_request_amount} {finalValue} {primaryCurrencyInfo.displayDenomination.name} ({secondaryCurrencyInfo.displayDenomination.symbol}
+            {fiat})
+          </T>
+        </View>
+        <View style={styles.textContainer}>
+          <T style={styles.text}>{s.strings.fio_request_requesting_to}</T>
+          <T style={styles.text}>{this.props.fioModalData.fioAddress}</T>
+        </View>
+        <View style={styles.selectContainer}>
+          <FormFieldSelect
+            style={MaterialInputStyle}
+            onChangeText={this.handleFioWalletChange}
+            label={`${s.strings.fio_request_requesting_from} ${s.strings.fio_address_confirm_screen_label}`}
+            value={selectedFioAddress}
+            labelExtractor={this.labelFromItem}
+            valueExtractor={this.labelFromItem}
+            data={walletAddresses.map(({ fioAddress }) => ({ fioAddress, label: fioAddress }))}
```

Comment:
Map function is pointless. Just change labelFromItem to use item.fioAddress

Reply:
changed